src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -3329,17 +3329,17 @@ int32 Unit::GetCurrentSpellCastTime(uint32 spell_id) const
 bool Unit::IsMovementPreventedByCasting() const
 {
     // can always move when not casting
-    if (!HasUnitState(UNIT_STATE_CASTING))
-        return false;
+    if (HasUnitState(UNIT_STATE_CASTING))
```

Comment:
No. you aren't checking attributes if you do this.

Reply:
is the second change correct? (seems like it to me)